action.yml

```diff
@@ -49,15 +49,16 @@ runs:
         DOCKER_BUILDKIT: '1'
         DOCKER_IO_USER: ${{ inputs.registry-username }}
         IMAGE_NAME: ${{ inputs.image_name }}
-        IMAGE_TAG: ${{ inputs.tags }}
 
       run: |
-        echo "::group::Cloning target branch..."
-        export BRANCH_NAME="${GITHUB_REF#refs/heads/}"
-        export BRANCH_NAME="${BRANCH_NAME#refs/tags/}"
-        echo "Current branch is: $BRANCH_NAME"
+        echo "::group::Cloning target ref..."
+        # initial ref is branch
+        export TARGET_REF="${GITHUB_REF#refs/heads/}"
+        # if tag, replace branch with tag
+        export TARGET_REF="${TARGET_REF#refs/tags/}"
```

Comment:
@fernandoataoldotcom if it's a tag could we also prevent it from pushing up a new version. Ex. it checks to see if one exists and just fails if the tag is already in the registry. This way we don't overwrite old tags by accident. I can throw it into jira if the change is a heavy lift

Reply:
for branches: if we're tagging with branch name and they create another commit in the context of the branch, don't we want to overwrite?

for tags - this just takes place in the context of a release, github won't allow a duplicate name

Reply:
This is a bit of an edge case where someone publishes something and someone later reruns the job publish a new artifact. For things like python/pip which may have additional unpinned dependencies it may create a moving target where one build is different than the next.

https://github.com/antoniostacos/wordpress/actions/runs/6463579649

Reply:
yes - you can manually trigger them, you just can't create a new release with the same tag

this also happens because we rebuild the container on every push, because the build may change between PR open and tag for deploy

probably the best way to solve this is to promote the same container build through all of the steps and add tags along the way because even the same SHA will potentially have different package dependencies

Reply:
thinking about this more, we probably want to change how we're handling builds / tags altogether
where we just build the image on PR open or new commits to an open PR and then just add tags as it is merged and released

I'll work on that update next